testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -4110,6 +4110,47 @@ public void testMergeAllInserts()
         assertUpdate("DROP TABLE " + targetTable);
     }
 
+    @Test
+    public void testMergeFalseJoinCondition()
+    {
+        skipTestUnless(hasBehavior(SUPPORTS_MERGE));
+
+        String targetTable = "merge_inserts_" + randomTableSuffix();
+        assertUpdate(createTableForWrites(format("CREATE TABLE %s (customer VARCHAR, purchases INT, address VARCHAR)", targetTable)));
+
+        assertUpdate(format("INSERT INTO %s (customer, purchases, address) VALUES ('Aaron', 11, 'Antioch'), ('Bill', 7, 'Buena')", targetTable), 2);
+
+        // Test a literal false
+        assertUpdate(format("MERGE INTO %s t USING ", targetTable) +
+                        "(VALUES ('Carol', 9, 'Centreville')) AS s(customer, purchases, address)" +
+                        " ON (FALSE)" +
+                        "    WHEN NOT MATCHED THEN INSERT (customer, purchases, address) VALUES(s.customer, s.purchases, s.address)",
+                1);
```

Comment:
You can use multiline strings for these tests:

```java
 assertUpdate("""
        MERGE INTO %s t USING 
            (VALUES ('Carol', 9, 'Centreville')) AS s(customer, purchases, address)
        ON (FALSE)
        WHEN NOT MATCHED THEN 
            INSERT (customer, purchases, address) VALUES(s.customer, s.purchases, s.address)
    """.formatted(targetTable),
    1);
```

Reply:
Changed.